torch/distributed/tensor/parallel/api.py

```diff
@@ -271,7 +272,7 @@ def _parallelize_linear(
             input_fn=parallel_style._prepare_input,  # type: ignore[arg-type, misc] # pyre-ignore[6]
             output_fn=parallel_style._prepare_output,  # type: ignore[arg-type, misc] # pyre-ignore[6]
         )
-    elif isinstance(parallel_style, ColwiseParallel):
+    elif isinstance(parallel_style, ColwiseParallel) or isinstance(parallel_style, ColwiseParallelNoReshard):
```

Comment:
this can be simplified to:
elif isinstance(parallel_style, (ColwiseParallel, ColwiseParallelNoReshard)):